src/StripedBackupSet.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AgDatabaseMove
+{
+
+  public class StripedBackupSet
+  {
+    public IEnumerable<BackupMetadata> StripedBackups { get; private set; }
+
+    private StripedBackupSet(IEnumerable<BackupMetadata> stripedBackups)
+    {
+      StripedBackups = stripedBackups;
+    }
+
+    public static IEnumerable<StripedBackupSet> GetStripedBackupSetChain(IEnumerable<BackupMetadata> backups)
+    {
+      var chain = backups
+        .GroupBy(b => new
+        {
+          b.FirstLsn,
+          b.LastLsn,
+          b.CheckpointLsn,
+          b.DatabaseBackupLsn,
+          b.BackupType,
+          b.DatabaseName
+        })
+        .Select(group => new StripedBackupSet(group.ToList()));
+      return chain;
```

Comment:
We can make this even better if we create a new custom `IEqualityComparer` class - maybe named `StripedBackupMetadataComparator`
@alexirion10 what do you think?

```csharp
var chain = backups
        .GroupBy(b => b, new StripedBackupMetadataComparator())
        .Select(group => new StripedBackupSet(group.ToList()));
```

Reply:
Either way sounds fine.  
What is the point of https://github.com/factset/AgDatabaseMove/blob/master/src/BackupMetadata.cs#L13 then?  Can you consolidate into a single comparator?

I don't think you need to call `ToList` here.  You're passing in an IEnumerable

Reply:
> Can you consolidate into a single comparator?

Yes that function will now be the `Equals()` function of the new comparator

> I don't think you need to call ToList here. You're passing in an IEnumerable

the `group` is an `IGrouping` which is an `IEnumerable` - correct
But it stores it as a dict if I don't do `ToList()`. But, I guess that should still be ok as long as it can enumerate over the stripes. 
(One of the benefits of using `IEnumerable`)
ref: https://docs.microsoft.com/en-us/dotnet/api/system.linq.igrouping-2?view=net-5.0

Reply:
Created a new `StripedBackupEqualityComparer`